server/queue/queue-api/src/test/java/org/apache/james/queue/api/PriorityMailQueueContract.java

```diff
@@ -0,0 +1,225 @@
+            .extracting(Mail::getName)
+            .containsExactly(NAME9, NAME8, NAME7, NAME6, NAME5, NAME4, NAME3, NAME2, NAME1, NAME0);
+    }
+
+    @Test
+    default void negativePriorityShouldBeConsideredZero() throws Exception {
+        getMailQueue().enQueue(mailBuilder()
+            .name(NAME2)
+            .attribute(MailPrioritySupport.MAIL_PRIORITY, -1)
+            .build());
+        getMailQueue().enQueue(mailBuilder()
+            .name(NAME1)
+            .attribute(MailPrioritySupport.MAIL_PRIORITY, 1)
+            .build());
+
+        MailQueue.MailQueueItem mailQueueItem1 = getMailQueue().deQueue();
+        mailQueueItem1.done(true);
+        MailQueue.MailQueueItem mailQueueItem2 = getMailQueue().deQueue();
+        mailQueueItem2.done(true);
+        assertThat(mailQueueItem1.getMail().getName()).isEqualTo(NAME1);
+        assertThat(mailQueueItem2.getMail().getName()).isEqualTo(NAME2);
+    }
+
+    @Test
+    default void tooBigPriorityShouldBeConsideredMaximum() throws Exception {
+        getMailQueue().enQueue(mailBuilder()
+            .name(NAME2)
+            .attribute(MailPrioritySupport.MAIL_PRIORITY, 12)
+            .build());
+        getMailQueue().enQueue(mailBuilder()
+            .name(NAME1)
+            .attribute(MailPrioritySupport.MAIL_PRIORITY, 8)
+            .build());
+
+        MailQueue.MailQueueItem mailQueueItem1 = getMailQueue().deQueue();
+        mailQueueItem1.done(true);
+        MailQueue.MailQueueItem mailQueueItem2 = getMailQueue().deQueue();
+        mailQueueItem2.done(true);
+        assertThat(mailQueueItem1.getMail().getName()).isEqualTo(NAME2);
+        assertThat(mailQueueItem2.getMail().getName()).isEqualTo(NAME1);
+    }
+
+    @Test
+    default void invalidPriorityShouldBeConsideredDefault() throws Exception {
+        getMailQueue().enQueue(mailBuilder()
+            .name(NAME1)
+            .attribute(MailPrioritySupport.MAIL_PRIORITY, "invalid")
+            .build());
+        getMailQueue().enQueue(mailBuilder()
+            .name(NAME2)
+            .attribute(MailPrioritySupport.MAIL_PRIORITY, 4)
+            .build());
+        getMailQueue().enQueue(mailBuilder()
+            .name(NAME3)
+            .attribute(MailPrioritySupport.MAIL_PRIORITY, 6)
+            .build());
+
+        MailQueue.MailQueueItem mailQueueItem1 = getMailQueue().deQueue();
+        mailQueueItem1.done(true);
+        MailQueue.MailQueueItem mailQueueItem2 = getMailQueue().deQueue();
+        mailQueueItem2.done(true);
+        MailQueue.MailQueueItem mailQueueItem3 = getMailQueue().deQueue();
+        mailQueueItem3.done(true);
+        assertThat(mailQueueItem1.getMail().getName()).isEqualTo(NAME3);
+        assertThat(mailQueueItem2.getMail().getName()).isEqualTo(NAME1);
+        assertThat(mailQueueItem3.getMail().getName()).isEqualTo(NAME2);
+    }
+
+    @Test
+    default void defaultPriorityShouldBeNormal() throws Exception {
```

Comment:
why does this mean ?

Reply:
It's more a **ShouldHaveNormalPriority**